### PR TITLE
done

### PR DESCRIPTION
**Pull request by khorkovenko**

### CODE REVIEW

File: store/src/main/java/com/book/store/config/AppConfig.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    public DataSource getDataSource() {
+        BasicDataSource basicDataSource = new BasicDataSource();
+        basicDataSource.setDriverClassName("org.h2.Driver");
+        basicDataSource.setUrl("jdbc:h2:mem:testdb");
+        basicDataSource.setUsername("sa");
+        basicDataSource.setPassword("password");
+
+        basicDataSource.setMinIdle(5);
+        basicDataSource.setMaxIdle(20);
+        basicDataSource.setMaxOpenPreparedStatements(100);
```
Comment (Sanderfuck) [inline store/src/main/java/com/book/store/config/AppConfig.java]: What about using properties file?

File: store/src/main/java/com/book/store/config/AppConfig.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    public LocalSessionFactoryBean getSessionFactory() {
+        LocalSessionFactoryBean localSessionFactoryBean = new LocalSessionFactoryBean();
+        localSessionFactoryBean.setDataSource(getDataSource());
+
+        Properties properties = new Properties();
+        properties.put("show_sql", "true");
+        properties.put("hibernate.hbm2ddl.auto", "create-drop");
+        localSessionFactoryBean.setHibernateProperties(properties);
+
+        localSessionFactoryBean.setPackagesToScan("com.book.store.model");
```
Comment (Sanderfuck) [inline store/src/main/java/com/book/store/config/AppConfig.java]: Use properties file

File: store/src/main/java/com/book/store/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    private String author;
+
+    @Column(unique = true)
+    private String isbn;
+
+    private BigDecimal price;
+
+    private String description;
+
+    private String coverImage;
```
Comment (Sanderfuck) [inline store/src/main/java/com/book/store/model/Book.java]: Validation?

File: store/src/main/java/com/book/store/repo/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    @Override
+    public List<Book> findAll() {
+        try {
+            Session session = sessionFactory.getCurrentSession();
+            return session.createQuery("FROM Book", Book.class)
+                    .getResultList();
+        } catch (PersistenceException e) {
+            throw new RuntimeException("Error finding all books", e);
+        }
+    }
```
Comment (Sanderfuck) [inline store/src/main/java/com/book/store/repo/impl/BookRepositoryImpl.java]: Read operation can use try-with-resources

File: store/pom.xml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-context</artifactId>
+			<version>6.0.11</version>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-orm</artifactId>
+			<version>6.0.11</version>
+		</dependency>
```
Comment (Sanderfuck) [inline store/pom.xml]: Dou you need it?

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: store/src/main/java/com/book/store/StoreApplication.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+public class StoreApplication {
+
+    public static void main(String[] args) {
+        SpringApplication.run(StoreApplication.class, args);
+    }
+
+    @Bean
+    public CommandLineRunner initializeData(BookService bookService) {
+        return args -> {
+            System.out.println("Jeff Bezos");
```
Comment (koval-kn) [inline store/src/main/java/com/book/store/StoreApplication.java]: it's not data initializing

File: store/mvnw.cmd
Comment (koval-kn) [inline store/mvnw.cmd:1]: we can exclude this file from git

File: store/mvnw
Comment (koval-kn) [inline store/mvnw:1]: we can exclude this file from git

File: store/.mvn/wrapper/maven-wrapper.properties
Comment (koval-kn) [inline store/.mvn/wrapper/maven-wrapper.properties:1]: we can exclude this file from git

File: store/.mvn/wrapper/maven-wrapper.jar
Comment (koval-kn) [inline store/.mvn/wrapper/maven-wrapper.jar:1]: we can exclude this file from git

File: .gitignore
Comment (koval-kn) [inline .gitignore:1]: consider using single .gitignore file

**Review by koval-kn — APPROVED** [review]

**Pull request closed without merging**